### PR TITLE
Soft voting

### PR DESCRIPTION
**Pull request by ricsuh**
Used VotingClassifier class from sklearn to make soft vote predictions.
Also, trained all learners for gender and msi on main.py.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -30,15 +37,28 @@
...
 #All of the other models that are not being trained here for msi right now should be added to this array.
 print("\n\n******************************************************************\nHARD VOTE FOR MSI")
-modelArrayMSI = [knn_msi, lr_msi]
+modelArrayMSI = [knn_msi, lr_msi, rf_msi, sgd_msi, nc_msi, mlp_msi]
 hvMSI = hv.hard_vote(modelArrayMSI, data.proteomic, lr_msi, 'msi')
 print(hvMSI)
 print("******************************************************************")
+
+# Soft Voting
+print("\n\n******************************************************************\n)
```
Comment (MSBradshaw) [inline main.py]: this string does not end, it is a syntax error

File: main.py [hunk 2]
```diff
@@ -30,15 +37,28 @@
...
+print("SOFT VOTE FOR GENDERS")
+gender_estimators = [('knn', knn_gender), ('lr', lr_gender), ('rf', rf_gender), ('sgd', sgd_gender), ('nc', nc_gender), ('mlp', mlp_gender)]
+sv_gender = sv.soft_vote(gender_estimators, data.proteomic, gender_labels)
+print("******************************************************************")
+
+print("\n\n******************************************************************\n)
```
Comment (MSBradshaw) [inline main.py]: syntax error

**Review by MSBradshaw — CHANGES_REQUESTED** [review]

File: main.py (hunk 1 above)
Comment (ricsuh) [reply main.py]: 1. Fixed syntax error. 
2. SGD: Apparently, probability estimates aren't available for hinge loss, which is SGD's default loss function. So I made a separate, modified function in learner_functions that changes the parameter in SGDClassifier() from `loss='hinge'` to `loss='modified_huber'`. This new loss function allows probability estimates.
3. NC: I removed Nearest Centroid from the list of learners because it doesn't predict probabilities, so it can't be used for soft voting (at least from what I've read on the internet...).

**Review by MSBradshaw — APPROVED** [review]
